onnxmltools/utils/metadata_props.py

```diff
@@ -0,0 +1,67 @@
+from ..proto import onnx, onnx_proto
+from ..convert.common.utils import version_string_to_tuple
+
+
+def _get_case_insensitive(iterable, key):
+    return next((x for x in iterable if x.lower() == key.lower()), None)
+
+
+def _validate_metadata(metadata_props):
+    valid_metadata_props = {
+        'Image.BitmapPixelFormat': ['Gray8', 'Rgb8', 'Bgr8', 'Rgba8', 'Bgra8'],
+        'Image.ColorSpaceGamma': ['Linear', 'SRGB'],
+        'Image.NominalPixelRange': ['NominalRange_0_255', 'Normalized_0_1', 'Normalized_1_1', 'NominalRange_16_235'],
+    }
+    image_metadata_props = {k: v for k, v in metadata_props.items() if _get_case_insensitive(valid_metadata_props, k)}
+    for key, value in image_metadata_props.items():
+        key = _get_case_insensitive(valid_metadata_props, key)
+        if not key:
+            print('Warning: key {} is defined multiple times'.format(key))
+        else:
+            valid_values = valid_metadata_props.pop(key)
+            if not _get_case_insensitive(valid_values, value):
+                print('Warning: value {} is invalid. Valid values are {}'.format(value, valid_values))
+    if image_metadata_props and valid_metadata_props:
+        print('Warning: incomplete image metadata is being added. Keys {} are missing.'.format(', '.join(valid_metadata_props)))
+
+COLOR_SPACE_TO_PIXEL_FORMAT = {
+    'BGR': 'Bgr8',
+    'RGB': 'Rgb8',
+    'GRAY': 'Gray8',
+}
+
+
+def color_space_to_pixel_format(color_space):
+    return COLOR_SPACE_TO_PIXEL_FORMAT[color_space]
+
+
+def add_metadata_props(onnx_model, metadata_props, targeted_onnx=onnx.__version__):
+    if version_string_to_tuple(targeted_onnx) < (1, 2, 1) or onnx_model.ir_version < 3:
```

Comment:
Are sure the value of onnx_model.ir_version was already assigned on calling add_metadata_props.